contracts/vanchor/linkable_merkle_tree.rs

```diff
@@ -0,0 +1,187 @@
+use crate::vanchor;
+use ink_prelude::vec::Vec;
+use ink_storage::traits::SpreadAllocate;
+#[cfg(feature = "std")]
+use ink_storage::traits::StorageLayout;
+use ink_storage::traits::{PackedLayout, SpreadLayout};
+use ink_storage::Mapping;
+use scale::{Decode, Encode, EncodeLike, Error, Input};
+
+pub type ChainId = u64;
+pub type Element = [u8; 32];
+pub type LatestLeafIndex = u32;
+
+pub const ROOT_HISTORY_SIZE: u32 = 100;
+
+#[derive(Default, Debug, Clone, SpreadLayout, PackedLayout)]
+#[cfg_attr(feature = "std", derive(StorageLayout, scale_info::TypeInfo))]
+pub struct Edge {
+    /// chain id
+    pub chain_id: ChainId,
+    /// root of source chain anchor's native merkle tree
+    pub root: [u8; 32],
+    /// height of source chain anchor's native merkle tree
+    pub latest_leaf_index: u32,
+    /// Target contract address or tree identifier
+    pub target: Element,
+}
+
+impl Encode for Edge {}
+
+impl EncodeLike for Edge {}
+
+impl Decode for Edge {
```

Comment:
Why not derive encode/decode traits instead? This shouldn't be left like this imo